course3/homework-string/my_string.h

```diff
@@ -1,6 +1,64 @@
 #pragma once
 
+#include <cstring>
+#include <memory>
+
 class MyString
 {
-	// TODO
+private:
+	std::unique_ptr<char[]> mBuffer;
+
+public:
+	MyString(): mBuffer(new char[1]) {}
```

Comment:
1. you should explicitly init the last (and first here) character to 0, otherwise it is left uninitialized.
2. nice container should not do any allocation in default constructor. Allocation may be expensive, so why do it to store nothing?